snapshots/output/syntax/src/destructuring.ts

```diff
@@ -0,0 +1,97 @@
+//                   ^^^^^ reference syntax 1.0.0 src/`destructuring.ts`/props.
+    return [props].map(({ a }) => a + b)
+//          ^^^^^ reference syntax 1.0.0 src/`destructuring.ts`/props.
+//                 ^^^ reference typescript 4.8.4 lib/`lib.es5.d.ts`/Array#map().
+//                        ^ definition local 10
+//                        documentation ```ts\n(parameter) a: number\n```
+//                        ^ reference syntax 1.0.0 src/`destructuring.ts`/Props#a.
+//                                ^ reference local 10
+//                                    ^ reference local 4
+  }
+  
+  export function arrayDestructuring(): number[] {
+//                ^^^^^^^^^^^^^^^^^^ definition syntax 1.0.0 src/`destructuring.ts`/arrayDestructuring().
+//                documentation ```ts\nfunction arrayDestructuring(): number[]\n```
+    const [b] = [props]
+//         ^ definition local 15
+//         documentation ```ts\nvar b: Props\n```
+//               ^^^^^ reference syntax 1.0.0 src/`destructuring.ts`/props.
+    return [[b]].map(([a]) => a.a)
+//           ^ reference local 15
+//               ^^^ reference typescript 4.8.4 lib/`lib.es5.d.ts`/Array#map().
+//                     ^ definition local 21
+//                     documentation ```ts\n(parameter) a: Props\n```
+//                            ^ reference local 21
+//                              ^ reference syntax 1.0.0 src/`destructuring.ts`/Props#a.
+  }
+  
+  export function nestedDestructuring(): number[] {
+//                ^^^^^^^^^^^^^^^^^^^ definition syntax 1.0.0 src/`destructuring.ts`/nestedDestructuring().
+//                documentation ```ts\nfunction nestedDestructuring(): number[]\n```
+    const [[b]] = [[props]]
+//          ^ definition local 28
+//          documentation ```ts\nvar b: Props\n```
+//                  ^^^^^ reference syntax 1.0.0 src/`destructuring.ts`/props.
+    return [[props]].map(([{ a }]) => a + b.a)
+//           ^^^^^ reference syntax 1.0.0 src/`destructuring.ts`/props.
+//                   ^^^ reference typescript 4.8.4 lib/`lib.es5.d.ts`/Array#map().
+//                           ^ definition local 36
+//                           documentation ```ts\n(parameter) a: number\n```
+//                           ^ reference syntax 1.0.0 src/`destructuring.ts`/Props#a.
+//                                    ^ reference local 36
+//                                        ^ reference local 28
+//                                          ^ reference syntax 1.0.0 src/`destructuring.ts`/Props#a.
+  }
+  
+  export function forLoopObjectDestructuring(): number {
+//                ^^^^^^^^^^^^^^^^^^^^^^^^^^ definition syntax 1.0.0 src/`destructuring.ts`/forLoopObjectDestructuring().
+//                documentation ```ts\nfunction forLoopObjectDestructuring(): number\n```
+    for (const { a } of [props]) {
+//               ^ definition local 41
+//               documentation ```ts\nvar a: number\n```
```

Comment:
This being a `var` and not a `const`/`let` is a bug, right? Filed https://github.com/sourcegraph/scip-typescript/issues/186

Reply:
Looks like a bug indeed, thank you for opening that